db/seeds.rb

```diff
@@ -5,3 +5,60 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+require 'open-uri'
+require 'nokogiri'
+puts "🔫 Destroy everything ..."
+# User.destroy_all
+# Products.destroy_all
+
+puts "🧑 Seed hardcoded users ..."
+barcode_sample = [4901330574352, 4954835290708, 4582469493006, 4902508070546, 4902777088785, 4902106010067, 4976548100075]
+# https://www.jancode.xyz/4901330574352/
+# https://www.jancode.xyz/4954835290708/
+
+
+barcode_sample = [4901330574352]
+barcode_sample.each do |barcode|
+  url = "https://www.jancode.xyz/#{barcode}/"
+  html_file = URI.open(url).read
+  html_doc = Nokogiri::HTML(html_file)
+  links = []
+  product_info = {}
+  test = []
+
+  html_doc.search('.keni-section h2').each do |element|
+    product_info["name"] = element.text.strip
+  end
+  # p product_info["name"]
+  html_doc.search('.table-block td span').each do |element|
+    product_info["barcode"] = element.text.strip
+  end
+
+  html_doc.search('.table-block tr:nth-child(8) td').each do |element|
```

Comment:
Seems like the row select processes around here are based on column index. It might cause some problem like some product doesn't have a specific column (for example this code '4978316991731' doesn't have an ingredient row so the 8th child is something else instead). Maybe you could select element by tr and use th as a column name (then check if the Japanese characters matched with our pattern (like 内容量 = size). Then if th(column header) is matched, then use the value from td in the same tr. But it works for now if you want to test for only 1 product sample.